Parsers/ASimDns/Parsers/ASimDnsInfobloxBloxOne.yaml

```diff
@@ -0,0 +1,227 @@
+      "IXFR", 251,        
+      "MAILB", 253,        
+      "MAILA", 254,        
+      "ANY", 255,            
+      "URI", 256,          
+      "CAA", 257,          
+      "TA", 32768,         
+      "DLV", 32769         
+  ];
+  let DnsResponseCodeLookup = datatable(EventResultDetails:string, DnsResponseCode:int)
+      [
+      "NOERROR", 0,        
+      "FORMERR", 1,        
+      "SERVFAIL", 2,       
+      "NXDOMAIN", 3,       
+      "NOTIMPL", 4,        
+      "REFUSED", 5,        
+      "YXDOMAIN", 6,       
+      "YXRRSET", 7,        
+      "NXRRSET", 8,        
+      "NOTAUTH", 9,        
+      "NOTZONE", 10,       
+      "DSOTYPENI", 11,     
+      "RESERVED12", 12,
+      "RESERVED13", 13,
+      "RESERVED14", 14,
+      "RESERVED15", 15,
+      "BADVERS", 16,       
+      "BADKEY", 17,        
+      "BADTIME", 18,       
+      "BADMODE", 19,       
+      "BADNAME", 20,       
+      "BADALG", 21,        
+      "BADTRUNC", 22,      
+      "BADCOOKIE", 23,     
+      ];
+  let parser = (disabled:bool=false) {
+      CommonSecurityLog
+      | where not(disabled) and DeviceVendor == "Infoblox" and DeviceEventClassID has "DNS"
+      | parse-kv AdditionalExtensions as (InfobloxDNSRCode:string, InfobloxDNSQType:string) with (pair_delimiter=";", kv_delimiter="=")
+      | project-rename 
+          EventResultDetails = InfobloxDNSRCode,
+          DnsQueryTypeName = InfobloxDNSQType
+      | extend DnsQueryTypeName = tostring(split(DnsQueryTypeName, ' ')[0])
+      | lookup EventSeverityLookup on LogSeverity
+      | lookup DnsQueryTypeLookup on DnsQueryTypeName
+      | lookup DnsResponseCodeLookup on EventResultDetails
+      | invoke _ASIM_ResolveDvcFQDN('DeviceName')
+      | project-rename 
+          DnsQuery = DestinationDnsDomain,
+          DvcIpAddr = DeviceAddress,
+          SrcIpAddr = SourceIP,
+          EventMessage = Message,
+          EventOriginalSeverity = LogSeverity,
+          EventOriginalType = Activity,
+          SrcUsername = SourceUserName,
+          SrcPortNumber = SourcePort
+      | extend
+          Dvc = coalesce(DvcHostname, DvcIpAddr),
+          EventEndTime = TimeGenerated,
+          EventResult = iff(EventResultDetails == "NOERROR", "Success", "Failure"),
+          DnsQuery = iff(substring(DnsQuery, strlen(DnsQuery) - 1, 1) == ".", substring(DnsQuery, 0, strlen(DnsQuery) - 1), DnsQuery),
+          EventStartTime = TimeGenerated,
+          Src = SrcIpAddr,
+          SrcUsernameType = _ASIM_GetUsernameType(SrcUsername),
+          DnsResponseCodeName = EventResultDetails,
+          IpAddr = SrcIpAddr,
+          User = SrcUsername
+      | extend Domain = DnsQuery
+      | extend
+          EventCount = toint(1),
+          EventSchema = "Dns",
+          EventSchemaVersion = "0.1.7",
+          EventProduct = "BloxOne",
+          EventVendor = "Infoblox",
+          EventType = "Query",
+          DnsQueryClass = toint(1),
+          DnsQueryClassName = "IN"
```

Comment:
is it always going to be 'IN'? Can't we extract this from AdditionalExtensions?

![image](https://github.com/user-attachments/assets/dca46766-a029-4ac5-a390-6dee3559271c)